src/store/slice/auth/types.ts

```diff
@@ -0,0 +1,58 @@
+export interface SignupPayloadI {
+  firstname: string;
+  lastname: string;
+  email: string;
+  password: string;
+}
+
+export interface SignupResponseI {
+  result: {
+    profilePicture: any;
+    firstname: string;
+    lastname: string;
+    email: string;
+    isEmailVerified: boolean;
+    _id: string;
+    createdAt: string;
+    updatedAt: string;
+  };
+}
+
+export interface AuthI {
+  loading: boolean;
+  user: SignupResponseI["result"] | null;
```

Comment:
What does this mean? Either user will have result object or null. And now look what have you define in auth reducer!!!